src/robot.cpp

```diff
@@ -87,24 +87,32 @@ void Robot::pubTransform(const geo::Pose3D &pose, const MapConfig &mapconfig)
     pub_joints_ground_truth.publish(jointState);
     pub_joints_internal.publish(jointState);
 
+    // Calculate tf transform
+    tf2::Transform tf_map, tf_robot, tf_total;
+
+    tf_robot.setOrigin(tf2::Vector3(pose.t.x, pose.t.y, pose.t.z + 0.044));
+    tf_robot.setRotation(tf2::Quaternion(0, 0, pose.getYaw()));
+
+    tf_map.setOrigin(tf2::Vector3(mapconfig.mapOffsetX, mapconfig.mapOffsetY, 0));
+    tf_map.setRotation(tf2::Quaternion(0, 0, mapconfig.mapOrientation));
+
+
+
+    tf_total = tf_map * tf_robot;
+
     // Publish tf transform
     geometry_msgs::TransformStamped transformStamped;
     transformStamped.header.stamp = ros::Time::now();
     transformStamped.header.frame_id = "map";
     transformStamped.child_frame_id = "ground_truth/base_link";
-    transformStamped.transform.translation.x =   pose.t.x + cos(mapconfig.mapOrientation) * mapconfig.mapOffsetX 
-                                                          + sin(mapconfig.mapOrientation) * mapconfig.mapOffsetY;
-
-    transformStamped.transform.translation.y =   pose.t.y - sin(mapconfig.mapOrientation) * mapconfig.mapOffsetX
-                                                          + cos(mapconfig.mapOrientation) * mapconfig.mapOffsetY;
-
-    transformStamped.transform.translation.z =   pose.t.z + 0.044;
-    tf2::Quaternion q;
-    q.setRPY(0, 0, pose.getYaw() + mapconfig.mapOrientation);
-    transformStamped.transform.rotation.x = q.x();
-    transformStamped.transform.rotation.y = q.y();
-    transformStamped.transform.rotation.z = q.z();
-    transformStamped.transform.rotation.w = q.w();
+    transformStamped.transform.translation.x =   tf_total.getOrigin().x();
+    transformStamped.transform.translation.y =   tf_total.getOrigin().y();
+    transformStamped.transform.translation.z =   tf_total.getOrigin().z();
+
+    transformStamped.transform.rotation.x = tf_total.getRotation().x();
+    transformStamped.transform.rotation.y = tf_total.getRotation().y();
+    transformStamped.transform.rotation.z = tf_total.getRotation().z();
+    transformStamped.transform.rotation.w = tf_total.getRotation().w();
```

Comment:
I think tf would offer a conversion for this. For example through http://wiki.ros.org/tf2_geometry_msgs.